### PR TITLE
{Network} `az network lb probe`: Add warnings for `--probe-threshold`

### PR DESCRIPTION
**Pull request by necusjz**
**Related command**

az network lb probe create
az network lb probe update
az network lb frontend-ip update

**Description**

Resolves: https://github.com/Azure/azure-cli/issues/24604

Fix the bug mentioned in https://github.com/Azure/azure-cli/pull/24366#discussion_r1036051313

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Network

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 1]
```diff
@@ -4782,65 +4782,102 @@ def set_lb_outbound_rule(instance, cmd, parent, item_name, protocol=None, outbou
...
+    for item in lb["probes"]:
+        probe = dict()
+        probe["name"] = item.pop("name", None)
+        probe["interval_in_seconds"] = item.pop("intervalInSeconds", None)
+        probe["number_of_probes"] = item.pop("numberOfProbes", None)
+        probe["port"] = item.pop("port", None)
+        probe["probe_threshold"] = item.pop("probeThreshold", None)
+        probe["protocol"] = item.pop("protocol", None)
+        probe["request_path"] = item.pop("requestPath", None)
+        probes.append(probe)
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:4808]: If new probe name exists in lb['probles'], do we need to raise error?

Comment (kairu-ms) [reply src/azure-cli/azure/cli/command_modules/network/custom.py:4808]: What's the behavior before?

Comment (necusjz) [reply src/azure-cli/azure/cli/command_modules/network/custom.py:4808]: Previously, we have a warning handled by `upsert_to_collection`:
https://github.com/Azure/azure-cli/blob/df34f7033b514f1f52244aaae70189821d5aedc0/src/azure-cli-core/azure/cli/core/commands/__init__.py#L486-L491

**Review by kairu-ms — APPROVED** [review]

**Review by yanzhudd — APPROVED** [review]

**Review by kairu-ms — APPROVED** [review]

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**